source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.h

```diff
@@ -0,0 +1,162 @@
+   * invalidate all cache entries where the version ID does not match.
+   * @return the cache version ID.
+   */
+  std::array<char, 4> cacheVersionId() const { return cache_version_id_; }
+
+  /**
+   * the size of the serialized proto message capturing headers and metadata.
+   * @return the size in bytes.
+   */
+  size_t headerSize() const { return header_size_; }
+
+  /**
+   * the size of the serialized proto message capturing trailers.
+   * @return the size in bytes.
+   */
+  size_t trailerSize() const { return trailer_size_; }
+
+  /**
+   * the size of the http body of the cache entry.
+   * @return the size in bytes.
+   */
+  size_t bodySize() const { return body_size_; }
+
+  /**
+   * sets the size of the serialized http headers, plus key and metadata, in the header block.
+   * @param sz The size of the serialized headers, key and metadata.
+   */
+  void setHeadersSize(size_t sz) { header_size_ = sz; }
+
+  /**
+   * sets the size of the serialized trailers in the header block.
+   * @param sz The size of the serialized trailers.
+   */
+  void setTrailersSize(size_t sz) { trailer_size_ = sz; }
+
+  /**
+   * sets the size of the serialized body in the header block.
+   * @param sz The size of the body data.
+   */
+  void setBodySize(size_t sz) { body_size_ = sz; }
+
+  /**
+   * the offset from the start of the file to the start of the serialized headers proto.
+   * @return the offset in bytes.
+   */
+  static off_t offsetToHeaders() { return size(); }
+
+  /**
+   * the offset from the start of the file to the start of the body data.
+   * @return the offset in bytes.
+   */
+  off_t offsetToBody() const { return offsetToHeaders() + headerSize(); }
+
+  /**
+   * the offset from the start of the file to the start of the serialized trailers proto.
+   * @return the offset in bytes.
+   */
+  off_t offsetToTrailers() const { return offsetToBody() + bodySize(); }
+
+  /**
+   * is this a valid cache file header block for the current code version?
+   * @return True if the block's cache version id and file id match the current version.
+   */
+  bool isValid() const;
+
+private:
+  /**
+   * sets the fileId, a fixed value used to identify that this is a cache file.
+   * This is private so only the unit test can call it, as the file ID should be
+   * a compile-time constant in ordinary use.
+   * @param id The file ID to set.
+   */
+  void setFileId(std::array<char, 4> id) { file_id_ = id; }
+
+  /**
+   * sets the cacheVersionId, a value that should be consistent between versions
+   * of the file cache implementation.
+   * This is private so only the unit test can call it, as the version ID should be
+   * a compile-time constant in ordinary use.
+   * @param id The cache version ID to set.
+   */
+  void setCacheVersionId(std::array<char, 4> id) { cache_version_id_ = id; }
+
+  std::array<char, 4> file_id_;
+  std::array<char, 4> cache_version_id_;
+  size_t header_size_;
```

Comment:
you'll see that in most places Envoy style is to use uint64_t rather than size_t. 

Also I thought you were going to keep headers & trailers at 32 bits though I' not really that concerned about it.

Reply:
size_t is consistent with the AsyncFileHandle interface, which is what this connects to eventually.
They're still being serialized as 32 bit, but size_t for descriptive purpose and AsyncFileHandle-relation.

Reply:
For the record I personally am fine with size_t for these kinds of usages but that has not been the practice in Envoy. That's also in conflict with common patterns in Envoy then :)

But I do not see that called out specifically in the style guide. But you can see how waves of inconsistency across the codebase can be a concern. Let me check with other maintainers about whether we should try to clean that up and clarify in the style guide, or let it go.

Reply:
Well, trying changing it here because I think that's what would cause the Windows CI failure!